modulemd/include/modulemd-2.0/modulemd-packager-v3.h

```diff
@@ -722,4 +723,28 @@ ModulemdComponentRpm *
 modulemd_packager_v3_get_rpm_component (ModulemdPackagerV3 *self,
                                         const gchar *component_name);
 
+
+/**
+ * modulemd_packager_v3_convert_to_index:
+ * @self: (in): This #ModulemdPackagerV3 object.
+ * @error: (out): A #GError that will return the reason for a conversion error.
+ *
+ * Note: If buildopts (#ModulemdBuildopts) are in use in one or more build
+ * configurations in the #ModulemdPackagerV3 object @self, only the buildopts
+ * present in the first listed configuration (if any) will be applied to the
+ * #ModulemdModuleStreamV2 object in the returned index.
```

Comment:
I think I was unclear in the design document for how this function should work.

In our original StreamV3 plans, we were trying to avoid changing MBS behavior, so we were forced to convert the PackagerV3 into a single StreamV2 object with dependencies in the MSE style. This is the behavior we want to happen transparently if they read in a PackagerV3 document using the [`ModuleStream.read_file()`](https://fedora-modularity.github.io/libmodulemd/latest/ModulemdModuleStream.html#modulemd-module-stream-read-file) or [`ModuleStream.read_string()`](https://fedora-modularity.github.io/libmodulemd/latest/ModulemdModuleStream.html#modulemd-module-stream-read-string) functions. We will need to document this as a limitation of the `ModuleStream.read_*()` API and recommend the use of the `Modulemd.read_packager_*()` API instead (I think by deprecating the `ModuleStream.read_*()` functions in favor of their `Modulemd.read_packager_*()` counterparts).

However, if MBS will use new API here, we can (and should) just convert it into a ModuleIndex of multiple StreamV2 objects, one for each build configuration specified in the PackagerV3. This will allow us to have separate buildopts for each of them.

Reply:
ACK. I'll rename this function back to private function `modulemd_packager_v3_to_stream_v2_ext()` and create a new implantation of `modulemd_packager_v3_convert_to_index()` with that behavior.